smtp.go

```diff
@@ -47,11 +46,22 @@ func (m *SMTPEmailBackend) Init(conf map[string]string) error {
 
 	host, port, err := net.SplitHostPort(conf["SMTPAddress"])
 	if err != nil {
-		return err
-	}
-	if port == "" {
-		port = conf["SMTPPort"]
+		if e, ok := err.(*net.AddrError); ok && e.Err == "missing port in address" {
+			if conf["SMTPPort"] == "" {
+				return e
+			}
+			host = conf["SMTPAddress"]
```

Comment:
why do we need this? is it not getting populated in line 47 ? or because of the error we need it?

Reply:
We do need these, otherwise `host` and `port` will be blank further down.